app/app/page.js

```diff
@@ -20,45 +21,35 @@ export default function Dashboard() {
   const dateFromFormatted = dateFrom.toISOString().substring(0, 10);
   const dateToFormatted = dateTo.toISOString().substring(0, 10);
 
+  const fetcher = useFetcher(); // This fetcher handles the token revalidation
+
   // Constructing the URL for fetching data, including host, token, and date range
   const endpointUrl = new URL(
     "/v0/pipes/ranking_of_top_organizations_creating_signatures.json",
     TINYBIRD_HOST
   );
-  endpointUrl.searchParams.set("token", TINYBIRD_TOKEN);
   endpointUrl.searchParams.set("date_from", dateFromFormatted);
   endpointUrl.searchParams.set("date_to", dateToFormatted);
 
   // Initializes variables for storing data
   let ranking_of_top_organizations_creating_signatures, latency, errorMessage;
 
-  try {
-    // Function to fetch data from Tinybird URL and parse JSON response
-    const fetcher = (url) => fetch(url).then((r) => r.json());
-
-    // Using SWR hook to handle state and refresh result every five seconds
-    const { data, error } = useSWR(endpointUrl.toString(), fetcher, {
-      refreshInterval: REFRESH_INTERVAL_IN_MILLISECONDS,
-    });
+  // Using SWR hook to handle state and refresh result every five seconds
+  const { data } = useSWR(endpointUrl, fetcher, {
```

Comment:
```suggestion
  const { data } = useSWR(endpointUrl.toString(), fetcher, {
```

`endpointUrl ` is a URL type and the fetcher waits for a string.

Reply:
thank you!!